os/drivers/video/video.c

```diff
@@ -317,6 +317,9 @@ static int change_video_state(FAR video_upperhalf_t *priv, enum video_state_e ne
 		dma_container = video_framebuff_get_dma_container(&priv->video_inf.bufinf);
 		if (dma_container) {
 			ret = video_devops->set_buf(priv->dev, V4L2_BUF_TYPE_VIDEO_CAPTURE, dma_container->buf.m.userptr, dma_container->buf.length);
+			if (ret < OK) {
```

Comment:
can there be a case when return value is not updated and set_buf still fails ??
In that case can w ehave the default ret value in the start of this API as ERROR ?

Reply:
Could you please point out where such a case might arise?
`set_buf` is expected to always return an ERROR value in case of failure and OK on success.

Reply:
Also if you could make the default return value (at initialization) as ERROR so once this call succeeds, it will be updated to OK, else ERROR value will be returned

Reply:
Changes made.